contracts/token/BALLEv2.sol

```diff
@@ -24,28 +27,44 @@ contract BALLEv2 is ERC20 {
     }
 
     /**
-     * @dev Set the new governance address
+     * @dev Modifier to check the caller is the governance address
      */
-    function setGovernance(address _governance) external {
+    modifier onlyGovernance() {
         require(msg.sender == governance, "!governance");
+        _;
+    }
+
+    /**
+     * @dev Modifier to check the caller is the governance address or an authorized minter
+     */
+    modifier onlyMinter() {
+        require(msg.sender == governance || minters[msg.sender], "!governance && !minter");
+        _;
+    }
+
+    /**
+     * @dev Set the new governance address
+     */
+    function setGovernance(address _governance) external onlyGovernance {
+        require(_governance != address(0), "zero address");
         governance = _governance;
         emit SetGovernance(_governance);
     }
 
     /**
      * @dev Add a minter address
      */
-    function addMinter(address _minter) external {
-        require(msg.sender == governance, "!governance");
+    function addMinter(address _minter) external onlyGovernance {
+        require(_minter != address(0), "zero address");
```

Comment:
I saw that you use constant for strings in the tests, thinking about do the same to avoid typos here, like 
const ZERO_ADDRESS = 'zero address';
does it make sense?

Reply:
Yeah, could be used, but this is the error message returned when the require fails, could be different for each function.